test/tools/utils.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+import fs from 'fs';
+import path from 'path';
+import {Float16Array} from '@petamoriken/float16';
+
+import {Tensor, sizeOfShape} from '../../src/lib/tensor.js';
+import {neg} from '../../src/unary.js';
+import {transpose} from '../../src/transpose.js';
+
+
+/**
+ * Convert data as required precision type.
+ * @param {Array<Number>} input
+ * @param {String} precisionType
+ * @return {Array<Number>}
+ */
+function getPrecisionData(input, precisionType) {
+  let data;
+  const isNumber = typeof input === 'number';
+  if (isNumber) {
+    input = [input];
+  }
+  switch (precisionType) {
+    case 'float32':
+      data = new Float32Array(input);
+      break;
+    case 'float16':
+      data = new Float16Array(input);
+      break;
+    case 'float64':
+      data = new Float64Array(input);
+      break;
+    case 'uint8':
+      data = new Uint8Array(input);
+      break;
+    case 'uint32':
+      data = new Uint32Array(input);
+      break;
+    case 'int64':
+      // data = new BigInt64Array(input.map(x => BigInt(x)));
+      // data = new Array(input.map(x => BigInt(x).toString()));
+      data = new Int32Array(input);
+      break;
+    default:
+      break;
+  }
+  if (isNumber) {
+    data = data[0];
+  }
+  return data;
+}
+
+
+/**
+ * Get converted data from given data dict with specified field and precision type.
+ * @param {Object} srcDataDict
+ * @param {String} source
+ * @param {String} precisionType
+ * @return {Array<Number>}
+ */
+function getPrecisionDataFromDataDict(srcDataDict, source, precisionType) {
+  const feedData = srcDataDict[source];
+  return getPrecisionData(feedData, precisionType);
+}
+
+/**
+ * Get a random number between the specified values
+ * @param {Number} min
+ * @param {Number} max
+ * @param {String} type default 'float64'
+ * @param {String} sign default 'positive'
+ * @return {Number}
+ */
+function getRandom(min, max, type = 'float64', sign = 'positive') {
+  if (sign === 'positive') {
+    if (min < 0) {
+      min = 0;
+    }
+  } else if (sign === 'negative') {
+    if (max > 0) {
+      max = 0;
+    }
+  }
+
+  if (type === 'float64' || type === 'float32' || type === 'float16') {
+    return Math.random() * (max - min) + min;
+  } else if (type === 'int32') {
+    min = Math.ceil(min) + 1;
```

Comment:
```suggestion
    min = Math.ceil(min);
```
Looks fishy. If you pass a minimum integer of 10, then the code will use 11 instead.

Reply:
Done.